File: pset6/readability/readability.py
from cs50 import get_string

text = get_string("Text: ")
textList = list(text)
letters = 0
sentences = 0
words = 1
index = 0

length = len(text)
for i in range(0,length):
    # check if it is a sentence
    if (text[i] == '.' or text[i] == '!' or text[i] == '?'):
        sentences += 1
    # check if it is a word
    if text[i] == ' ':
        words += 1
    if text[i].isalpha():
        letters += 1

L = float(letters / words) * 100;
S = float(sentences / words) * 100;
index = (0.0588 * L) - (0.296 * S) - 15.8;

if index < 1:
    print("Before Grade 1")
elif index > 16:
    print("Grade 16+")
else:
    print("Grade " + str(round(index)))

File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void){
    string text = get_string("Text: ");
    int letters = 0;
    int sentences = 0;
    int words = 1;
    float index = 0;
    for (int i = 0, n = strlen(text); i < n ; i ++){
        //check if is a sentence
        if (text[i] == '.' || text[i] == '!' || text[i] =='?'){
            sentences += 1;
        }
        //check if it is a word
        if (isspace(text[i])){
            words += 1;
        }
        //count the amount of letters
        if (isalpha(text[i])){
            letters += 1;
        }
    }
    
    //Coleman-Liau index
    float L = ((float)letters / words) * 100;
    float S = ((float)sentences / words) * 100;
    index = (0.0588 * L) - (0.296 * S) - 15.8;
    
    if (index < 1){
        printf("Before Grade 1\n");
    }
    else if (index > 16){
        printf("Grade 16+\n");
    }
    else{
        printf("Grade %i\n", (int) round(index));
    }
}